cst438_registration_backend/cst438_registration/src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -84,34 +84,40 @@ public String createStudent(@RequestBody StudentDTO studentDTO) {
 	/*
 	 * Update a student
 	 */
-	@PutMapping("/student/update/{student_id}")
-	public void updateStudent(@PathVariable("id") int id, @RequestBody StudentDTO studentDTO) {
-		Student student = studentRepository.findByStudent_id(id);
-		if (student == null) {
-			throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Student not found.");
-		}
-		
+	/*
+	 * Update a student
+	 */
+	@PutMapping("/student/{student_email}")
+	public void updateStudent(@RequestBody StudentDTO studentDTO) {
 		List<Student> students = (List<Student>) studentRepository.findAll();

```

Comment:
Could maybe use Optional class to only pull student with correct email instead of entire dataset?